jwst/pathloss/pathloss.py

```diff
@@ -335,18 +340,29 @@ def do_correction(input_model, pathloss_model):
             ymax = int(y.max())
             ra, dec, wavelength = slice_wcs(x, y)
             wavelength_array[ymin:ymax+1, xmin:xmax+1] = wavelength
-        pathloss_pointsource_2d = interpolate_onto_grid(wavelength_array,
-                                                        wavelength_pointsource,
-                                                        pathloss_pointsource_vector)
-        pathloss_uniformsource_2d = interpolate_onto_grid(wavelength_array,
-                                                          wavelength_uniformsource,
-                                                          pathloss_uniform_vector)
-        input_model.pathloss_pointsource2d = pathloss_pointsource_2d
-        input_model.pathloss_uniformsource2d = pathloss_uniformsource_2d
-        #
+
+        # Compute and apply the 2D correction
+        if is_pointsource(input_model.meta.target.source_type):
+            pathloss_pointsource_2d = interpolate_onto_grid(
+                wavelength_array,
+                wavelength_pointsource,
+                pathloss_pointsource_vector)
+            output_model.data /= pathloss_pointsource_2d
+            output_model.err /= pathloss_pointsource_2d
+            output_model.var_poisson /= pathloss_pointsource_2d**2
+        else:
+            pathloss_uniformsource_2d = interpolate_onto_grid(
+                wavelength_array,
+                wavelength_uniformsource,
+                pathloss_uniform_vector)
+            output_model.data /= pathloss_uniformsource_2d
+            output_model.err /= pathloss_uniformsource_2d
+            output_model.var_poisson /= pathloss_uniformsource_2d**2
+
         # This might be useful to other steps
-        input_model.wavelength = wavelength_array
-        input_model.meta.cal_step.pathloss = 'COMPLETE'
+        output_model.wavelength = wavelength_array
```

Comment:
Looks like this wavelength array is saved to the output only for NIRSpec IFU mode, correct? I don't think we create a wavelength array for MIRI MRS exposures, so that's inconsistent for the 2 IFU's. @jemorrison is this wavelength array that's in the NIRSpec IFU 2-D slice images ever used anywhere downstream?

Reply:
Yeah, I don't think we need to save this wavelength array.  Unless it gets recomputed down the line and would be useful to have.  @jemorrison?  @nden?